dbt/adapters/athena/impl.py

```diff
@@ -231,7 +231,12 @@ def get_glue_table(self, relation: AthenaRelation) -> Optional[GetTableResponseT
             glue_client = client.session.client("glue", region_name=client.region_name, config=get_boto3_config())
 
         try:
-            table = glue_client.get_table(CatalogId=catalog_id, DatabaseName=relation.schema, Name=relation.identifier)
+            if catalog_id is not None:
```

Comment:
is it possible to fix `get_catalog_id` method instead ? we can get current account id using `sts` client I gues. Does it make sense ?

Reply:
Ah good suggestion. I'll have a look!

Reply:
Ah, this already happens: https://github.com/dbt-athena/dbt-athena/blob/main/dbt/adapters/athena/impl.py#L535-L539

Reply:
I started to look why my code is actually calling `_get_data_catalog` with `database=None`. 
It's because my custom macro (insert-by-period) is creating a relation from scratch:

```jinja
{%- set tmp_relation = api.Relation.create(identifier=tmp_identifier, schema=schema, type='table') -%}
```

However, you can pass `schema` **and/or** `database` in the `Relation.create` method ([see docs](https://docs.getdbt.com/reference/dbt-classes#creating-relations)). However, in dbt-athena's `impl.py` it tends to mix `relation.schema` and `relation.database` throughout the code. Any idea why we do that? 
In my case, `relation.database` ends up None, but code that calls `relation.schema` works 🤷‍♂️ . 

dbt-athena will work with `relation.database` to get the data catalog: 

https://github.com/dbt-athena/dbt-athena/blob/9dec391257fce15b1887d8fa6bb8dd4b7e8261a2/dbt/adapters/athena/impl.py#L227

But dbt-athena uses `relation.schema` in other places, even in the same function 🤔 :

https://github.com/dbt-athena/dbt-athena/blob/9dec391257fce15b1887d8fa6bb8dd4b7e8261a2/dbt/adapters/athena/impl.py#L234


A fix on my side, would be to define both schema and database.

```jinja
{%- set tmp_relation = api.Relation.create(identifier=tmp_identifier, schema=schema, database=schema, type='table') -%}
```

As I understood, in athena the schema is the same as database. So they are mixed in usage. But maybe it's good to consistently work with schema **or** database in the dbt-athena codebase?

Reply:
The concept of relation.database in dbt is similar to what in athena/trino is called catalog, and what are databases in glue are schemas in dbt.

Therefore to me seems correct to use `relation.schema` as database in the APIs calls - but the database must be the catalog not the schema right?

Reply:
Right 👍 database = datacatalog, schema = database 😓 

I fixed the dbt-athena v1.6.1 issues by specifying `database` too in the [insert-by-period](https://gist.github.com/jessedobbelaere/6fdb593f9e2cc732e9f142c56c9bac87) macro custom relation. It was missing the database field:

```sql
{%- set tmp_relation = api.Relation.create(database=target_relation.database, schema=target_relation.schema, identifier=tmp_identifier, type='table') -%}
```

Not sure what to do now with this PR now? Close it? 

In practice, it should never have a database=None
The default aws account Id is already implemented in the code:

https://github.com/dbt-athena/dbt-athena/blob/9dec391257fce15b1887d8fa6bb8dd4b7e8261a2/dbt/adapters/athena/impl.py#L535-L539